TileStache/Goodies/VecTiles/transform.py

```diff
@@ -2067,47 +2137,51 @@ def remove_duplicate_features(
     # features. we keep the geometry of the seen items too, so that
     # we can tell if any new feature is significantly far enough
     # away that it should be shown again.
-    seen_items = dict()
-
-    def meters_to_pixels(distance):
-        return distance * float(1 << (zoom + 8)) / 40075016.68
-
-    new_features = []
-    for feature in layer['features']:
-        shape, props, fid = feature
-
-        keep_feature = True
-        if shape.geom_type in geometry_types:
-            key = tuple([props.get(k) for k in property_keys])
-            seen_geoms = seen_items.get(key)
-
-            if seen_geoms is None:
-                # first time we've seen this item, so keep it in
-                # the output.
-                seen_items[key] = [shape]
+    deduplicator = _Deduplicator(property_keys, min_distance,
+                                 none_means_unique)
+
+    for source_layer in source_layers:
+        layer_index = -1
+        # because this post-processor can potentially modify
+        # multiple layers, and that wasn't how the return value
+        # system was designed, instead it modifies layers
+        # *in-place*. this is abnormal, and as such requires a
+        # nice big comment like this!
```

Comment:
Should we just change the contract so that post processors can modify any number of layers? Maybe they should just return the new list?